server/src/main/java/io/crate/planner/optimizer/Rule.java

```diff
@@ -33,7 +34,12 @@
 
     Pattern<T> pattern();
 
-    LogicalPlan apply(T plan, Captures captures, TableStats tableStats, TransactionContext txnCtx, NodeContext nodeCtx);
+    LogicalPlan apply(T plan,
+                      Captures captures,
+                      TableStats tableStats,
+                      TransactionContext txnCtx,
```

Comment:
We could introduce a RuleContext, as an `Uber-Context` including TransactionContext, NodeContext and PlannerContext.